include/gatekeeper_gk.h

```diff
@@ -122,6 +140,13 @@ struct gk_fib {
 			 */
 			struct ether_cache eth_cache;
 		} grantor;
+
+		/*
+		 * When the action is GK_FWD_BACK_NET or GK_FWD_FRONT_NET,
+		 * we need the next fib entry for the gateway of the
+		 * destination.
+		 */
+		struct gk_fib *next_fib;
```

Comment:
Yes for the first question.

Give your correct observation, you can use DPDK's hash table for both: IPv4 and IPv6.